src/javascript/_common/base/login.js

```diff
@@ -19,30 +22,36 @@ const Login = (() => {
         const language   = getLanguage();
         const signup_device      = LocalStore.get('signup_device') || (isMobile() ? 'mobile' : 'desktop');
         const date_first_contact = LocalStore.get('date_first_contact');
-        const marketing_queries   = `&signup_device=${signup_device}${date_first_contact ? `&date_first_contact=${date_first_contact}` : ''}`;
+        const marketing_queries  = `&signup_device=${signup_device}${date_first_contact ? `&date_first_contact=${date_first_contact}` : ''}`;
 
         return ((server_url && /qa/.test(server_url)) ?
             `https://${server_url}/oauth2/authorize?app_id=${getAppId()}&l=${language}${marketing_queries}` :
             urlForCurrentDomain(`https://oauth.binary.com/oauth2/authorize?app_id=${getAppId()}&l=${language}${marketing_queries}`)
         );
     };
 
-    const isLoginPages = () => /logged_inws|redirect/i.test(window.location.pathname);
-
-    const socialLoginUrl = brand => (`${loginUrl()}&social_signup=${brand}`);
+    const socialLoginUrl = (brand, affiliate_token, utm_source, utm_medium, utm_campaign) => (
+        `${loginUrl()}&social_signup=${brand}${affiliate_token}${utm_source}${utm_medium}${utm_campaign}`
+    );
 
     const initOneAll = () => {
         ['google', 'facebook'].forEach(provider => {
             $(`#button_${provider}`).off('click').on('click', e => {
                 e.preventDefault();
-                window.location.href = socialLoginUrl(provider);
+
+                const utm_data     = TrafficSource.getData();
+                const utm_source   = TrafficSource.getSource(utm_data) ? `&utm_source=${TrafficSource.getSource(utm_data)}` : '';
```

Comment:
better store `TrafficSource.getSource(utm_data)` in a variable so we don't call it twice